neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -140,6 +140,7 @@ public virtual bool Reverify(Snapshot snapshot, IEnumerable<Transaction> mempool
             fee += mempool.Where(p => p != this && p.Sender.Equals(Sender)).Select(p => (BigInteger)(p.SystemFee + p.NetworkFee)).Sum();
             if (balance < fee) return false;
             UInt160[] hashes = GetScriptHashesForVerifying(snapshot);
+            if (hashes.Length != Witnesses.Length) return false;
```

Comment:
This has been verified in `VerifyWitnesses()`. It is impossible that they are unequal.

Reply:
No, `Reverify` executed before `VerifyWitnesses`

```csharp
public virtual bool Verify(Snapshot snapshot, IEnumerable<Transaction> mempool)
        {
            if (!Reverify(snapshot, mempool)) return false;
            int size = Size;
            if (size > MaxTransactionSize) return false;
            long net_fee = NetworkFee - size * NativeContract.Policy.GetFeePerByte(snapshot);
            if (net_fee < 0) return false;
            return this.VerifyWitnesses(snapshot, net_fee);
        }
```